idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/symbols/KtFirSymbolProvider.kt

```diff
@@ -109,9 +106,10 @@ internal class KtFirSymbolProvider(
         }
     }
 
-    override fun getNamedClassOrObjectSymbol(psi: KtClassOrObject): KtNamedClassOrObjectSymbol = withValidityAssertion {
+    override fun getNamedClassOrObjectSymbol(psi: KtClassOrObject): KtNamedClassOrObjectSymbol? = withValidityAssertion {
         require(psi !is KtObjectDeclaration || psi.parent !is KtObjectLiteralExpression)
-        psi.withFirDeclarationOfType<FirRegularClass, KtNamedClassOrObjectSymbol>(resolveState) {
+        // A KtClassOrObject may also map to an FirEnumEntry. Hence, we need to return null in this case.
+        psi.withFirDeclarationOfTypeOrNull<FirRegularClass, KtNamedClassOrObjectSymbol>(resolveState) {
```

Comment:
Instead of introducing new `withFirDeclarationOfTypeOrNull`, let's just check here that `psi !is KtEnumEntry`. Otherwise, we implicitly hidding error when non-enum KtClassOrObject was mapped to some invalid FirElement

Reply:
Sounds good. Thanks!